src/diffpy/utils/tests/test_strumining_helpers.py

```diff
@@ -0,0 +1,42 @@
+from diffpy.utils.pdfitc_helpers import strumining_processing
+from diffpy.utils.parsers import load_PDF_into_db, markup_PDF, apply_schema
+from diffpy.utils.parsers import loadData
+from diffpy.utils.tests.testhelpers import datafile
+
+import os
+import filecmp
+
+tests_dir = os.path.dirname(os.path.abspath(locals().get('__file__', 'file.py')))
+
+nload = datafile("loadneutron.txt")
+xload = datafile("loadxray.txt")
+
+neutron_target = datafile("targetsmn.json")
+xray_target = datafile("targetsmx.json")
+
+neutron_generated = datafile("gensmn.json")
+xray_generated = datafile("gensmx.json")
+
+sm_schema = datafile("strumining.json")
+
+
+def test_sm_preprocessing():
+    # simple workflow for neutron data
+    hddata, rv = loadData(nload, headers=True)
+    hddata = strumining_processing(hddata)
+    markup_PDF(neutron_generated, hddata, rv)
+    apply_schema(neutron_generated, sm_schema)
+
+    # simple workflow for xray data
+    hddata, rv = loadData(xload, headers=True)
+    hddata = strumining_processing(hddata)
+    markup_PDF(xray_generated, hddata, rv)
+    apply_schema(xray_generated, sm_schema)
+
+    # check against target files
+    assert filecmp.cmp(neutron_generated, neutron_target)
+    assert filecmp.cmp(xray_generated, xray_target)
+
+    # cleanup
+    os.remove(neutron_generated)
```

Comment:
this is not a great way to handle things in tests.   Use the tmpfile pytest fixture.    This creates a temporary directory and you can write things there, then read them and do hte asserts etc.   This is then created and removed cleanly even if hte tests fail or error (at least if I understand correclty what you are doing here).

Reply:
Got it, that's what I was going for

Reply:
this is how regolith does it to make a fs database backend. This is defined in conftest.py and passed into hte tests
```
@pytest.fixture(scope="session")
def make_db():
    """A test fixutre that creates and destroys a git repo in a temporary
    directory.
    This will yield the path to the repo.
    """
    cwd = os.getcwd()
    name = "regolith_fake"
    repo = os.path.join(tempfile.gettempdir(), name)
    if os.path.exists(repo):
        rmtree(repo)
    subprocess.run(["git", "init", repo])
    os.chdir(repo)
    with open("README", "w") as f:
        f.write("testing " + name)
    with open("regolithrc.json", "w") as f:
        json.dump(
            {
                "default_user_id": "sbillinge",
                "groupname": "ERGS",
                "databases": [
                    {
                        "name": "test",
                        "url": repo,
                        "public": True,
                        "path": "db",
                        "local": True,
                        "backend": "filesystem"
                    }
                ],
                "repos": [
                    {"_id": "talk_repo",
                     "params": {
                         "namespace_id": "35",
                         "initialize_with_readme": "true",
                         "name": "repo name"},
                     "url": "https://example.com",
                     "api_route": "/url/example",
                     "namespace_name": "talks"
                     }
                ],
                "stores": [
                    {
                        "name": "store",
                        "url": repo,
                        "path": repo,
                        "public": True,
                    }
                ],
                "tokens": [
                    {
                        "_id": "gitlab_private_token",
                        "token": "<private_token>"
                    }
                ],
            },
            f,
        )
    fspath = os.path.join(repo, 'db')
    os.mkdir(fspath)
    exemplars_to_fs(fspath)
    subprocess.run(["git", "add", "."])
    subprocess.run(["git", "commit", "-am", "Initial readme"])
    yield repo
    os.chdir(cwd)
    if not OUTPUT_FAKE_DB:
        rmtree(repo)
```

Reply:
Instead of using this, pytest allows a tmp_path that is cleaned up even upon failures (confirmed with false assertions).